src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1015,3 +1057,917 @@ def upgrade_agents(cmd, client, resource_group_name, cluster_name, kube_config=N
+                if replace or i == j:
+                    existing[key].remove(j)
+                else:
+                    msg = 'A different object named {} already exists in your kubeconfig file.\nOverwrite?'
+                    overwrite = False
+                    try:
+                        overwrite = prompt_y_n(msg.format(i['name']))
+                    except NoTTYException:
+                        pass
+                    if overwrite:
+                        existing[key].remove(j)
+                    else:
+                        msg = 'A different object named {} already exists in {} in your kubeconfig file.'
+                        telemetry.set_exception(exception='A different object with same name exists in the kubeconfig file', fault_type=consts.Different_Object_With_Same_Name_Fault_Type,
+                                                summary=msg.format(i['name'], key))
+                        raise CLIError(msg.format(i['name'], key))
+        existing[key].append(i)
+
+
+def _resolve_service_principal(client, identifier):  # Uses service principal graph client
+    # todo: confirm with graph team that a service principal name must be unique
+    result = list(client.list(filter="servicePrincipalNames/any(c:c eq '{}')".format(identifier)))
+    if result:
+        return result[0].object_id
+    if utils.is_guid(identifier):
+        return identifier  # assume an object id
+    error = CLIError("Service principal '{}' doesn't exist".format(identifier))
+    error.status_code = 404  # Make sure CLI returns 3
+    raise error
+
+
+def client_side_proxy_wrapper(cmd,
+                              client,
+                              resource_group_name,
+                              cluster_name,
+                              token=None,
+                              path=os.path.join(os.path.expanduser('~'), '.kube', 'config'),
+                              overwrite_existing=False,
+                              context_name=None,
+                              api_server_port=consts.API_SERVER_PORT):
+
+    client_proxy_port = consts.CLIENT_PROXY_PORT
+    signal.signal(signal.SIGINT, ctrlc_handler)
+
+    cloud = send_cloud_telemetry(cmd)
+    args = []
+    operating_system = platform.system()
+    proc_name = f'arcProxy{operating_system}'
+
+    telemetry.set_debug_info('CSP Version is ', consts.CLIENT_PROXY_VERSION)
+    telemetry.set_debug_info('OS is ', operating_system)
+
+    if(check_process(proc_name)):
+        telemetry.set_user_fault()
+        raise CLIError('Another instance of proxy already running')
+
+    port_error_string = ""
+    if check_if_port_is_open(api_server_port):
+        port_error_string += f'Port {api_server_port} is already in use. Please select a different port with --api-server option.\n'
+    if check_if_port_is_open(client_proxy_port):
+        telemetry.set_exception(exception='Client proxy port was in use.', fault_type=consts.Client_Proxy_Port_Fault_Type,
```

Comment:
set user fault here

Reply:
we are raising a user fault in the below if. So not needed here.